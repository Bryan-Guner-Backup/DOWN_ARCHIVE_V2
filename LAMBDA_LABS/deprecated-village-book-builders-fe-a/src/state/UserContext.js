import React, { useState, useContext, useEffect } from 'react';

const UserContext = React.createContext();

export function useUser() {
  return useContext(UserContext);
}

export function UserProvider({ children }) {
  const [userInfo, setUserInfo] = useState({
    username: localStorage.getItem('username'),
    role: localStorage.getItem('role'),
    id: localStorage.getItem('id'),
  });

  useEffect(() => {
    if (
      !localStorage.getItem('username') === null ||
      !localStorage.getItem('role') === null
    ) {
      return;
    } else {
      localStorage.setItem('username', userInfo.username);
      localStorage.setItem('role', userInfo.role);
      localStorage.setItem('id', userInfo.id);
    }
  }, [userInfo]);
  return (
    <UserContext.Provider value={{ userInfo, setUserInfo }}>
      {children}
    </UserContext.Provider>
  );
}
