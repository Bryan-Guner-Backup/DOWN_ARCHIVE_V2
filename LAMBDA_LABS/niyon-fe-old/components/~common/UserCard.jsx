import React from 'react';
import styled from 'styled-components';
import Avatar from './Avatar';
import Icon from './Icon';
import Text from './Text';

const Card = styled.div`
  width: ${({ width }) => width || '250px'};
  box-shadow: ${({ theme }) => theme.boxShadow};
  border-radius: ${({ theme }) => theme.borderRadius};
  margin: 0 10px;
`;

const Row = styled.div`
  margin: 8px 8px 5px 5px;
  display: flex;
  justify-content: flex-start;
`;

const Info = styled.div`
  flex-grow: 1;
  min-width: 0;
  margin: 0 2px;
`;

const P = styled.p`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin: 3px;
  font-size: ${({ theme }) => theme.smallText};

  &.name {
    font-weight: ${({ theme }) => theme.bold};
  }
`;

function UserCard(props) {
  const { source, name, jobTitle, city, country } = props;

  return (
    <Card>
      <Row style={{ display: 'flex' }}>
        <div>
          <Avatar medium source={source}></Avatar>
        </div>
        <Info style={{}}>
          <P className="name">{name}</P>
          <P>{jobTitle}</P>
          <P className="location">
            <Icon small black className="fas fa-map-marker-alt" /> {city},{' '}
            {country}
          </P>
        </Info>
        <Icon medium className="fas fa-user-plus" black></Icon>
      </Row>
    </Card>
  );
}

export default UserCard;
