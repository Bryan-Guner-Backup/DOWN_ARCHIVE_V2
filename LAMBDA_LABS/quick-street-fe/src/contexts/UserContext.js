import React, { createContext, useState } from 'react';

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  
  const [ user, setUser ] = useState({
    userId: '',
    isVendor: '',
    token: ''
  
  });
  
  const addUser = (newUser) => {
    // console.log('new user ', newUser);
    setUser({
      ...user, 
      userId: newUser.id, 
      token: newUser.token, 
      isVendor: newUser.isVendor
    });
  }

  return (
    <UserContext.Provider value={{ user, addUser }}>
      {children}
    </UserContext.Provider>
  )
}
