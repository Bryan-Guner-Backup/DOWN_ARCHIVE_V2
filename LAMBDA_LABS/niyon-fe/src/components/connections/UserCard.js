import React, { useContext, useState } from 'react'
import Card from '@material-ui/core/Card'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import Avatar from '@material-ui/core/Avatar'
import AddBoxIcon from '@material-ui/icons/AddBox'
import { UserContext } from '../../UserContext'
import { getUserCard } from '../apiStuff/axiosWithAuth'
import Popper from '@material-ui/core/Popper'

const useStyles = makeStyles({
  root: {
    width: '100%',
    maxWidth: 500,
    textAlign: 'left',
    padding: '5px 10px'
  },
  paper: {
    border: '1px solid',
    borderRadius: '10px',
    height: '40px',
    width: '100px',
    backgroundColor: '#3f51b5'
  },
  popup: {
    textAlign: 'center',
    margin: 'auto 0',
    fontSize: 'medium',
    color: 'white',
    paddingTop: '8px'
  }
})

export default function UserCard (props) {
  const classes = useStyles()
  /*eslint-disable */
  const { user, setUser } = useContext(UserContext);
  const myRequests = user.myRequests;
  const id = window.localStorage.getItem("id");
  const payload =
    props.endpoint === "request"
      ? { mentor_id: props.value.id }
      : { status: true, rejected: false, userReq: props.value.id };
  const [anchorEl, setAnchorEl] = useState(null);
  const open = Boolean(anchorEl);
  const popUpId = open ? "transitions-popper" : undefined;

  const handleRequest = (event) => {
    const endPoint  = props.endpoint
    // axiosWithAuth()
    //   .post(`/connection/${props.endpoint}/${id}`, payload)
    getUserCard(endPoint, id, payload)
      .then(
        (res) => console.log(res),
        setAnchorEl(anchorEl ? null : event.currentTarget)
      )
      .catch((err) => console.log(err));
  };

  return (
    <React.Fragment>
      {myRequests && (
        <Card className="userCard">
          <div className="avatar">
            <Avatar src="/broken-image.jpg" />
          </div>
          <div className={classes.root}>
            <Typography variant="h4" gutterBottom>
              {props.value.first_name} {props.value.last_name}
            </Typography>
            <Typography variant="h6" gutterBottom>
              <p>{props.value.location}</p>
            </Typography>
            <Typography variant="h6" gutterBottom>
              <p>{props.value.job_title}</p>
            </Typography>
          </div>
          <div className="addIcon">
            <AddBoxIcon
              onClick={handleRequest}
              style={{ color: "green" }}
              className="iconSize"
            ></AddBoxIcon>
            <Popper id={popUpId} open={open} anchorEl={anchorEl}>
              <div className={classes.paper}>
                <p className={classes.popup}>Success!</p>
              </div>
            </Popper>
          </div>
        </Card>
      )}
      {/* {!myRequests && <p>Loading....</p>} */}
    </React.Fragment>
  );
}
