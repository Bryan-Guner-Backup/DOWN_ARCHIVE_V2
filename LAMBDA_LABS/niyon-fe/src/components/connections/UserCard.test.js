import React from 'react'
import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json';
import  UserCard  from './UserCard'
import { MemoryRouter } from 'react-router-dom'
import { UserContext } from '../../UserContext'
import findByTestAttr from '../../tests/utils'
import toJson from 'enzyme-to-json'

const testProps = {
  endpoint: "response",
  value: { 
          bio: "Learning code now.",
          email: "testReqest@aol.com",
          first_name: "test-fn",
          id: 142,
          job_title: "Full Stack Web Developer",
          job_title_id: 2,
          last_name: "test-ln",
          location: "test-1, test-2",
          location_id: 2,
          techs: (2) [1, 9],
          user_type: "test-request-Mentor"
        } 
}

const testUser = {
  bio: "learning code",
  email: "testUser@aol.com",
  first_name: "test-user-fn",
  id: 143,
  job_title: "Full Stack Web Developer",
  job_title_id: 2,
  last_name: "test-user-ln",
  location: "test-user-area-1, test-user-area-2",
  location_id: 2,
  myConnections: [testProps],
  myRequests: [],
  mySentRequests: [],
  techs: [4],
  user_type: "test-usr-Mentor",
}

describe('<UserCard /> component', () => {
  const user = testUser
  const setUser = jest.fn()
  let component
  const profiles = []
  const handleClick = jest.fn()
  const id = window.localStorage.getItem('id')
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <UserContext.Provider value={{ user, setUser }}>        
          <UserCard {...testProps} />        
      </UserContext.Provider>
    )
    component = mount(
      <UserContext.Provider value={{ user, setUser }}>        
          <UserCard {...testProps} />        
      </UserContext.Provider>
    )
  })

  it('should pass snapshot testing', () => {
      expect(toJson(component)).toMatchSnapshot()
  })

  it('should render the component correctly', () => {
      expect(component.exists()).toBe(true);
  })

  it('should render Avatar image correctly',() => {
      // console.log(component.debug())
      expect(component.find('img').exists()).toBe(true)
      expect(component.find('img').length).toBe(1)
  })

  it('should render the user first name and last name correctly', () => {
      const fullName = `${testProps.value.first_name} ${testProps.value.last_name}`;
      expect(component.find('h4').exists()).toBe(true)
      expect(component.find('h4').length).toBe(1)
      expect(component.find('h4').text().trim()).toEqual(fullName)
  })
  it('should render user location correctly', () => {
      const location = testProps.value.location;
      const locationWrapper = component.find('h6').first();
      expect(locationWrapper.exists()).toBe(true);
      expect(locationWrapper.find('p').exists()).toBe(true);
      expect(locationWrapper.find('p').text().trim()).toEqual(location);
  })

  it('should render user job title correctly', () => {
    const job_title = testProps.value.job_title;
    const jobTitleWrapper = component.find('h6').at(1);
    expect(jobTitleWrapper.exists()).toBe(true);
    expect(jobTitleWrapper.find('p').exists()).toBe(true);
    expect(jobTitleWrapper.find('p').text().trim()).toEqual(job_title);
  })

  it('should render add icon div correctly', () => {
     expect(component.find('div.addIcon').length).toBe(1);
  })

  it('should send the add request when the user clicks on add button', () => {
      // const spyClick = jest.spyOn(component, 'handleRequest')
      // const addButton = component.find('div.addIcon').find('svg');
      const handleRequest = jest.fn();      
      const addButton = findByTestAttr(wrapper, 'add-box-icon');
      expect(addButton).toBeTruthy();
      // addButton.simulate('click')
      // expect(handleRequest).toHaveBeenCalled();
  })
    
})