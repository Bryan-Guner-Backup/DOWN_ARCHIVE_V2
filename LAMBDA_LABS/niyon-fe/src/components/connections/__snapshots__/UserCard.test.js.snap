// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> component should pass snapshot testing 1`] = `
<UserCard
  endpoint="response"
  value={
    Object {
      "bio": "Learning code now.",
      "email": "testReqest@aol.com",
      "first_name": "test-fn",
      "id": 142,
      "job_title": "Full Stack Web Developer",
      "job_title_id": 2,
      "last_name": "test-ln",
      "location": "test-1, test-2",
      "location_id": 2,
      "techs": undefined,
      "user_type": "test-request-Mentor",
    }
  }
>
  <WithStyles(ForwardRef(Card))
    className="userCard"
  >
    <ForwardRef(Card)
      className="userCard"
      classes={
        Object {
          "root": "MuiCard-root",
        }
      }
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiCard-root userCard"
        elevation={1}
      >
        <ForwardRef(Paper)
          className="MuiCard-root userCard"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={1}
        >
          <div
            className="MuiPaper-root MuiCard-root userCard MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="avatar"
            >
              <WithStyles(ForwardRef(Avatar))
                src="/broken-image.jpg"
              >
                <ForwardRef(Avatar)
                  classes={
                    Object {
                      "circle": "MuiAvatar-circle",
                      "colorDefault": "MuiAvatar-colorDefault",
                      "fallback": "MuiAvatar-fallback",
                      "img": "MuiAvatar-img",
                      "root": "MuiAvatar-root",
                      "rounded": "MuiAvatar-rounded",
                      "square": "MuiAvatar-square",
                    }
                  }
                  src="/broken-image.jpg"
                >
                  <div
                    className="MuiAvatar-root MuiAvatar-circle"
                  >
                    <img
                      className="MuiAvatar-img"
                      src="/broken-image.jpg"
                    />
                  </div>
                </ForwardRef(Avatar)>
              </WithStyles(ForwardRef(Avatar))>
            </div>
            <div
              className="makeStyles-root-1"
            >
              <WithStyles(ForwardRef(Typography))
                gutterBottom={true}
                variant="h4"
              >
                <ForwardRef(Typography)
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  gutterBottom={true}
                  variant="h4"
                >
                  <h4
                    className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
                  >
                    test-fn
                     
                    test-ln
                  </h4>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                gutterBottom={true}
                variant="h6"
              >
                <ForwardRef(Typography)
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  gutterBottom={true}
                  variant="h6"
                >
                  <h6
                    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                  >
                    <p>
                      test-1, test-2
                    </p>
                  </h6>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                gutterBottom={true}
                variant="h6"
              >
                <ForwardRef(Typography)
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  gutterBottom={true}
                  variant="h6"
                >
                  <h6
                    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                  >
                    <p>
                      Full Stack Web Developer
                    </p>
                  </h6>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
            </div>
            <div
              className="addIcon"
            >
              <ForwardRef
                className="iconSize"
                onClick={[Function]}
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                <WithStyles(ForwardRef(SvgIcon))
                  className="iconSize"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  <ForwardRef(SvgIcon)
                    className="iconSize"
                    classes={
                      Object {
                        "colorAction": "MuiSvgIcon-colorAction",
                        "colorDisabled": "MuiSvgIcon-colorDisabled",
                        "colorError": "MuiSvgIcon-colorError",
                        "colorPrimary": "MuiSvgIcon-colorPrimary",
                        "colorSecondary": "MuiSvgIcon-colorSecondary",
                        "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                        "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                        "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                        "root": "MuiSvgIcon-root",
                      }
                    }
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "green",
                      }
                    }
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root iconSize"
                      focusable="false"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "green",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                      />
                    </svg>
                  </ForwardRef(SvgIcon)>
                </WithStyles(ForwardRef(SvgIcon))>
              </ForwardRef>
              <ForwardRef(Popper)
                anchorEl={null}
                open={false}
              />
            </div>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</UserCard>
`;
